public/app/features/dashboard/components/TransformationsEditor/TransformationsEditor.tsx

```diff
@@ -12,45 +12,39 @@ interface Props {
 }
 
 interface State {
-  updateCounter: number;
+  addingTransformation: boolean;
 }
 
 export class TransformationsEditor extends React.PureComponent<Props, State> {
-  state = { updateCounter: 0 };
+  state = { addingTransformation: false };
 
-  onTransformationAdd = () => {
+  onTransformationAdd = (selectable: SelectableValue<string>) => {
     const { transformations, onChange } = this.props;
     onChange([
       ...transformations,
       {
-        id: DataTransformerID.noop,
+        id: selectable.value as string,
         options: {},
       },
     ]);
-    this.setState({ updateCounter: this.state.updateCounter + 1 });
+    this.setState({ addingTransformation: false });
   };
 
   onTransformationChange = (idx: number, config: DataTransformerConfig) => {
     const { transformations, onChange } = this.props;
     transformations[idx] = config;
```

Comment:
Doesn't this mutate state?

Reply:
Yes, it does. I will fix that one as well. Good catch 🙌